src/internal/common/readers/retry_handler.go

```diff
@@ -31,6 +34,19 @@ var retryErrs = []error{
 	io.ErrUnexpectedEOF,
 }
 
+// TODO(pandeyabs): Consolidate error funcs with retryErrs slice.
+func isTimeoutErr(err error) bool {
+	switch err := err.(type) {
+	case *url.Error:
+		return err.Timeout()
+	}
+
+	return errors.Is(err, context.Canceled) ||
+		errors.Is(err, context.DeadlineExceeded) ||
```

Comment:
can we skip the checks on context deadlines and cancellation? I feel like those are things we'll be wanting to handle separately since they could be either a backup that's hung for a while or the user trying to exit the backup early